### PR TITLE
Update to receive signals from docker

### PR DESCRIPTION
**Pull request by khoinguyen2992**
- Add more signals to terminate
- Add init kafka with broker version
- Fix mark offset to prevent messages duplication
- Fix deadlock when connected but got error at the first time

### CODE REVIEW

File: common.go [hunk 1]
```diff
@@ -3,19 +3,29 @@ package kafka
...
+	syscall.SIGQUIT,
+	syscall.SIGABRT,
+}
+
 func execFunction(msg string, funcs ...interface{}) (result []reflect.Value, err error) {
 	funcs = append(funcs, msg)
 	f := reflect.ValueOf(funcs[0])
 	params := funcs[1:]
-	fmt.Printf("All params(%d): %#v, %#v\n", len(params), params)
+	fmt.Printf("All params(%d): %#v\n", len(params), params)
```
Comment (tranduythanh) [inline common.go:29]: I think this line (and all fmt.Print... lines) should be deleted after debugging :)

File: kafka.go [hunk 2]
```diff
@@ -1,15 +1,41 @@
...
+	case "0.8.2.2":
+		return sarama.V0_8_2_2
+	case "0.9.0.0":
+		return sarama.V0_9_0_0
+	case "0.9.0.1":
+		return sarama.V0_9_0_1
+	case "0.10.0.0":
+		return sarama.V0_10_0_0
+	default:
+		panic("Version unsupported.")
```
Comment (tranduythanh) [inline kafka.go]: `Version unsupported` --> `Unsupported Version` 💃

File: consumer_group.go [hunk 3]
```diff
@@ -12,34 +12,37 @@ import (
...
+	Partitions  string
+	BrokerList  string
+	Offset      string
+	Verbose     bool
+	BufferSize  int
+	Messages    chan *sarama.ConsumerMessage
+	Closing     chan struct{}
+	WaitGroup   sync.WaitGroup
+	Version     sarama.KafkaVersion
+	DoneInitial bool
```
Comment (tranduythanh) [inline consumer_group.go]: How about the name: `ReadyState` or `Ready`? `DoneInitial` is not a good name in this case in my opinion

File: consumer_group.go [hunk 4]
```diff
@@ -12,34 +12,37 @@ import (
...
+		GroupID:     group,
+		Partitions:  this.Partitions,
+		BrokerList:  this.BrokerList,
+		Offset:      "newest",
+		Verbose:     false,
+		BufferSize:  this.BufferSize,
+		Messages:    make(chan *sarama.ConsumerMessage, defaultBufferSize),
+		Closing:     make(chan struct{}),
+		WaitGroup:   this.WaitGroup,
+		DoneInitial: false,
```
Comment (tranduythanh) [inline consumer_group.go]: As what I said above. I think, we should use the name: `ReadyState` or `Ready` too 💃

File: common.go (hunk 1 above)
Comment (khoinguyen2992) [inline common.go:29]: I think we should keep them for a while. Just in case.

**Pull request merged**